src/bin/btc-cold.rs

```diff
@@ -405,11 +415,23 @@ impl Args {
         Ok(())
     }
 
-    fn address(&self, path: &Path, count: u16, skip: u16, show_change: bool) -> Result<(), Error> {
+    fn address(
+        &self,
+        path: &Path,
+        count: u16,
+        skip: u16,
+        show_change: bool,
+        regtest: bool,
+    ) -> Result<(), Error> {
         let secp = Secp256k1::new();
 
         let file = fs::File::open(path)?;
         let descriptor: Descriptor<DerivationAccount> = Descriptor::strict_decode(file)?;
+        let target_network = if regtest {
+            Network::Regtest
```

Comment:
This will not work: if a user will provide a mainnet descriptor with mainnet keys and `--regtest` flag this will switch to the Regtest w/o notification, while it should just print an error message and fail to process